runtime/lua/vim/lsp.lua

```diff
@@ -112,6 +112,22 @@ function lsp._unsupported_method(method)
   return msg
 end
 
+---@private
+---@param workspace_folders string|lsp.WorkspaceFolder[]?
+---@return lsp.WorkspaceFolder[]?
+function lsp._get_workspace_folders(workspace_folders)
```

Comment:
Not entirely sure but I think a valid configuration should always have either a `root_dir` or `workspace_folders` set. If so, this function would return a non-null array.

Reply:
In the single file case one can have `nil` as `root_dir` and no workspace folders.

Reply:
I see.
And how does this PR break a single client with multiple single files?

Reply:
> And how does this PR break a single client with multiple single files?

I was thinking of running `:lua vim.lsp.start({ cmd = {"lua-language-server"}, name = "luals" })` twice (root dir being nil). 
But looks like it already starts a new client on each call without this change.